### PR TITLE
Oidc tokens documentation

### PR DESCRIPTION
**Pull request by Lexman42**

### CODE REVIEW

File: website/source/docs/secrets/identity/index.html.md [hunk 1]
```diff
@@ -134,6 +134,176 @@ group and has an alias that maps to the group in LDAP. If the user is removed
...
+
+### Verifying Authenticity of ID Tokens Generated by Vault
+
+An identity token may be verified by the client party using the public keys
+published by Vault, or via a Vault-provided introspection endpoint.
+
+Vault will serve standard "[.well-known](https://tools.ietf.org/html/rfc5785)"
+endpoints that allow easy integration with OIDC verification libraries.
+Configuring the libraries will typically involve providing an issuer URL and
+client ID. The library will then handle key requests and can validation the
```
Comment (ncabatoff) [inline website/source/docs/secrets/identity/index.html.md]: s/validation/validate/

File: website/source/docs/secrets/identity/index.html.md [hunk 2]
```diff
@@ -134,6 +134,176 @@ group and has an alias that maps to the group in LDAP. If the user is removed
...
+only requiring _access_ to Vault, not _authorization_, as the .well-known
+endpoints are unauthenticated.
+
+Alternatively, the token may be sent to Vault for verification via an
+[introspection endpoint](api/secret/identity/tokens.html#introspect-a-signed-id-token).
+The response will indicate whether the token is "active" or not, as well as any
+errors that occurred during validation. Beyond simply allowing the client to
+delegate verification to Vault, using this endpoint incorporates the additional
+check of whether the entity is still active or not, which is something that
+cannot be determined from the token alone. Unlike the .well-known, accessing the
```
Comment (ncabatoff) [inline website/source/docs/secrets/identity/index.html.md]: Either say "the .well-known endpoint" or simply ".well-known".

File: website/source/docs/secrets/identity/index.html.md [hunk 3]
```diff
@@ -134,6 +134,176 @@ group and has an alias that maps to the group in LDAP. If the user is removed
...
+
+
+### Issuer Considerations
+
+The identity token system has one configurable parameter: issuer. The issuer
+`iss` claim is particularly important for proper validation of the token by
+clients, and special consideration should be given when using Identity Tokens
+with [performance replication](docs/enterprise/replication/index.html).
+Consumers of the token will request public keys from Vault using the issuer URL,
+so it much be network reachable. Furthermore, the returned keyset will include
```
Comment (ncabatoff) [inline website/source/docs/secrets/identity/index.html.md]: s/much/must/

Comment (ncabatoff) [reply website/source/docs/secrets/identity/index.html.md]: This is the first use of the term of `keyset`. If not a typo, please define it.

File: website/source/docs/secrets/identity/index.html.md [hunk 4]
```diff
@@ -134,6 +134,176 @@ group and has an alias that maps to the group in LDAP. If the user is removed
...
+an issuer that must match the request.
+
+By default Vault will set the issuer to the Vault instance's
+[`api_addr`](docs/configuration/index.html#api_addr) This means that tokens
+issued in a given cluster should be validated within that same cluster.
+Alternatively, the [`issuer`](api/secret/identity/tokens.html#issuer) parameter
+may be configured explicitly. This address must point to the identity/oidc path,
```
Comment (ncabatoff) [inline website/source/docs/secrets/identity/index.html.md]: Is "the identity/oidc path" explicit enough?  Or should we spell it out a bit more?

Comment (kalafut) [reply website/source/docs/secrets/identity/index.html.md]: This brings up good point: should a custom issuer just be domain and port. We're appending `/v1/identity/oidc` to the api_addr, but using a provided issuer verbatim. Maybe we should append the path to both.

Comment (kalafut) [reply website/source/docs/secrets/identity/index.html.md]: I've created a task to address this (I think it should be changed), but will clarify the as-is docs now.

File: website/source/docs/secrets/identity/index.html.md [hunk 5]
```diff
@@ -134,6 +134,176 @@ group and has an alias that maps to the group in LDAP. If the user is removed
...
+"key" will be used to sign the token. The role template is an optional parameter
+to customizes the token contents and is described in the next section. Token TTL
+controls the expiration time of the token, after which verification library will
+consider the token invalid. All roles have a Vault-generated `client_id`
+attribute that is returned when the role is read. This value cannot be changed
+and will be added to the token's `aud` parameter. JWT/OIDC libraries will often
+require this value.
+
+A role's "key" parameter links a role to an existing named key. A single key may
+be used by multiple roles, and using multiple named keys is generally not
```
[the remark this answers is not in the document]
Comment (kalafut) [reply website/source/docs/secrets/identity/index.html.md]: Agreed.

**Pull request merged**